src/session_middleware.rs

```diff
@@ -68,13 +74,14 @@ where
 
       if let Some(jwt) = &jwt {
         // Ignore any invalid auth so the site can still be used
-        // TODO: this means it will be impossible to get any error message for invalid jwt. Need
-        //       to add a separate endpoint for that.
-        //       https://github.com/LemmyNet/lemmy/issues/3702
-        let local_user_view = local_user_view_from_jwt(jwt, &context).await.ok();
-        if let Some(local_user_view) = local_user_view {
-          req.extensions_mut().insert(local_user_view);
-        }
+        let local_user_id = Claims::validate(jwt, &context)
+          .await
+          .with_lemmy_type(LemmyErrorType::NotLoggedIn)?;
+        let local_user_view = LocalUserView::read(&mut context.pool(), local_user_id)
+          .await
+          .with_lemmy_type(LemmyErrorType::IncorrectLogin)?;
+        check_user_valid(&local_user_view.person)?;
+        req.extensions_mut().insert(local_user_view);
```

Comment:
The old code would quietly ignore any errors here (eg while reading LocalUserView). It would then throw an error [here](https://github.com/LemmyNet/lemmy/blob/main/crates/db_views/src/local_user_view.rs#L123) which has nothing to do with the original cause. So its better to fail early and make it clear what happened.